app/serializers/site_category_serializer.rb

```diff
@@ -36,4 +36,19 @@ def include_allow_global_tags?
   def include_required_tag_groups?
     SiteSetting.tagging_enabled
   end
+
+  def notification_level
+    return if !scope || scope.anonymous?
+
+    CategoryUser.where(user_id: scope.user.id, category_id: object.id).pick(:notification_level) ||
+      CategoryUser.default_notification_level
+  end
+
+  def permission
+    return if !scope || scope.anonymous?
```

Comment:
```suggestion
    return if scope.anonymous?
```

I don't think `scope` is ever blank in our serializers.

Reply:
It is, see `Site.all_categories_cache`.